Parser/workhorseParse.m

```diff
@@ -167,13 +168,42 @@
   
   %
   % calculate velocity (speed and direction)
-  % currently assuming earth coordinate transform
+  % check for ENU and Beam coordinates. Instrument coordinates not
+  % considered.
   %
-  
-  veast = velocity.velocity1;
-  vnrth = velocity.velocity2;
-  wvel  = velocity.velocity3;
-  evel  = velocity.velocity4;
+  adcpOrientations = str2num(fixed.systemConfiguration(:, 1)); % str2num is actually more relevant than str2double here
+  adcpOrientation = mode(adcpOrientations); % hopefully the most frequent value reflects the orientation when deployed
+%   if adcpOrientation == 1
+%       txt = 'UP';
+%   else
+%       txt = 'DOWN';
+%   end
+%   str = input(['ADCP ' num2str(fixed.instSerialNumber(1)) ' is oriented ' txt '. Return if this is correct, ''N'' to orient opposite direction: '],'s');
+%   if ~isempty(strmatch('N', upper(str)))
+%       if adcpOrientation == 1
+%           adcpOrientation = 0;
+%           adcpOrientations(adcpOrientations==1)=0;
+%       else
+%           adcpOrientation = 1;
+%           adcpOrientations(adcpOrientations==0)=1;
+%       end
+%   end
+      
+  if mode(fixed.coordinateTransform) == 31 %ENU
```

Comment:
Regards coordinate system type (ENU,BEAM), In local code for modified workhorseParse.m have comment/code 
```
  % EX/coordinate transform
  % xxx00xxx = NO TRANSFORMATION (BEAM COORDINATES)
  % xxx01xxx = INSTRUMENT COORDINATES
  % xxx10xxx = SHIP COORDINATES
  % xxx11xxx = EARTH COORDINATES
  % xxxxx1xx = TILTS (PITCH AND ROLL) USED IN SHIP OR EARTH TRANSFORMATION
  % xxxxxx1x = 3-BEAM SOLUTION USED IF ONE BEAM IS BELOW THE CORRELATION THRESHOLD SET BY THE WC command
  % xxxxxxx1 = BIN MAPPING USED
  sect.coordinateTransform = double(data(idx+25));
  cfg.coordinateSystem     = getopt(bitand(bitshift(mode(sect.coordinateTransform),-3),3),'BEAM','INSTRUMENT','SHIP','EARTH');
```
The idea from R. Pawlowicz rdradcp.m

Just want to highlight that the 7/31 magic numbers might be dependent on other bits set in ```sect.coordinateTransform```

And if all adcp's have at least the concept of BEAM and EARTH coordinates, then this configuration could be determined in the parsers and stored in sample_data.meta somehow (just equivalently amongst all adcp parsers).

Reply:
the mode should be forwarded in meta. The parser should not do the conversion - this is the work of PP routines.